code/game/objects/structures/floodlight.dm

```diff
@@ -0,0 +1,320 @@
+/obj/machinery/power/floodlight
+	name = "floodlight"
+	icon = 'icons/obj/floodlight.dmi'
+	icon_state = "floodlight0_hatchopen0"
+	desc = "An industrial floodlight. It has units of power remaining."
+	density = 1
+	anchored = 0
+	var/status = 0
+	var/obj/item/weapon/stock_parts/cell/powerpack
+	var/datum/effect_system/spark_spread/sparks = new
+	var/health = 60
+	var/maxhealth = 60
+	var/broken = 0
+	var/cover = 0
+	var/hascell = 0
```

Comment:
Indeed, the hasCell variable is redundant, could just check whether powerpack is null or not to determine if it has one.